### ISSUE TITLE
[java] JUnitSpelling false-positive for JUnit5/4 tests

### ISSUE THREAD

**Issue by KroArtem** [pmd/pmd#2477]
**Affects PMD Version:** 
6.23

**Description:**
[JUnitSpelling](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#junitspelling) requires some tweaks in terms of documentation and produces false positives for:
1) JUnit4 class when it doesn't extend `TestCase` but has a `setup` method
2) JUnit5 class when it has a `setup` method

**Code Sample demonstrating the issue:**

```java
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class BTest {

    @BeforeEach
    void setup() {    }

    @Test
    void test() {    }
}
```

```java
import org.junit.Before;
import org.junit.Test;
public class BTest {

    @Before
    void setup() {    }

    @Test
    void test() {    }
}
```

**Running PMD through:** *[CLI]*

I suggest two things:

1) Do not trigger if it's a junit5 class (`isJUnit5Class` from `AbstractJUnitRule` can be used).
2) Do not trigger if junit4 doesn't extend `TestCase`

WDYT?

**Comment by adangel** [comment]
Sounds good. It should make this rule less noisy if one has it enabled even if not needed. The rule is only useful, if you still use JUnit3.
